docs/core/extensions/sslstream-best-practices.md

```diff
@@ -0,0 +1,107 @@
+
+## Selecting cipher suites
+
+SslStream allows users to specify which cipher suites can be negotiated by the TLS handshake via the <xref:System.Net.Security.CipherSuitesPolicy> class. As with TLS versions, we recommend letting the operating system decide which are the best cipher suites to negotiate with and, therefore, we recommend avoiding the use of <xref:System.Net.Security.CipherSuitesPolicy>.
+
+Note that <xref:System.Net.Security.CipherSuitesPolicy> is not available on all platforms supported by .NET.
+
+## Specifying a Server Certificate
+
+When authenticating as a server, <xref:System.Net.Security.SslStream> requires an <xref:System.Security.Cryptography.X509Certificates.X509Certificate> instance to use. There are multiple ways how the server certificate can be passed to <xref:System.Net.Security.SslStream>:
+
+- Directly as a parameter to <xref:System.Net.Security.SslStream.AuthenticateAsServerAsync?displayProperty=nameWithType> or via <xref:System.Net.Security.SslServerAuthenticationOptions.ServerCertificate?displayProperty=nameWithType> property
+- From a selection callback in <xref:System.Net.Security.SslServerAuthenticationOptions.ServerCertificateSelectionCallback?displayProperty=nameWithType> property
+- By passing a <xref:System.Net.Security.SslStreamCertificateContext> in the <xref:System.Net.Security.SslServerAuthenticationOptions.ServerCertificateContext?displayProperty=nameWithType> property
+
+The recommended approach is to use the <xref:System.Net.Security.SslServerAuthenticationOptions.ServerCertificateContext?displayProperty=nameWithType> property. When the certificate is obtained by one of the other two ways, a <xref:System.Net.Security.SslStreamCertificateContext> instance is created internally by the <xref:System.Net.Security.SslStream> implementation. Creating a <xref:System.Net.Security.SslStreamCertificateContext> involves building an <xref:System.Security.Cryptography.X509Certificates.X509Chain> which is a CPU intensive operation. It is more efficient to create a <xref:System.Net.Security.SslStreamCertificateContext> once and reuse it for multiple <xref:System.Net.Security.SslStream> instances.
+
+Reusing <xref:System.Net.Security.SslStreamCertificateContext> instances also enables additional features such us [TLS session resumption](https://datatracker.ietf.org/doc/html/rfc5077) on Linux servers.
+
+## Custom X509Certificate validation
+
+There are certain scenarios in which the default certificate validation procedure is not adequate and some custom validation logic is required. This custom logic can be provided via the <System.Net.Security.SslClientAuthenticationOptions.RemoteCertificateValidationCallback> property. As an illustration, following sections provide some examples.
```

Comment:
We may also mention [CertificateChainPolicy](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslserverauthenticationoptions.certificatechainpolicy?view=net-7.0#system-net-security-sslserverauthenticationoptions-certificatechainpolicy) that was added in 7.0 and perhaps show validation with custom trust.